File: test_privy_stellar.py
#!/usr/bin/env python3
import requests
import base64
import json

# Privy credentials
app_id = 'cmhow02lw00b3l10cz7f0gbpu'
app_secret = '3hRZCYhv4CP9iRsT33GVD8TCtzJhAmooMaQ94CWvDXbwSS75wvgbKuCMbFLfLgCfacSRwxyfK11qq6jNjh3BCciE'

# Create Basic Auth header
credentials = base64.b64encode(f'{app_id}:{app_secret}'.encode()).decode()

# Test Stellar wallet creation
print("Testing Privy Stellar Wallet Creation...\n")

url = 'https://api.privy.io/v1/wallets'
headers = {
    'Content-Type': 'application/json',
    'Authorization': f'Basic {credentials}',
    'privy-app-id': app_id,
    'privy-idempotency-key': 'test_stellar_wallet_12345'
}

data = {
    'chain_type': 'stellar'
}

print("1. Creating Stellar wallet via Privy...")
response = requests.post(url, headers=headers, json=data)

print(f"Status Code: {response.status_code}")
print(f"Response: {json.dumps(response.json(), indent=2)}\n")

if response.status_code in [200, 201]:
    wallet_data = response.json()
    print("✅ Stellar wallet created successfully!")
    print(f"   Wallet ID: {wallet_data.get('id', 'N/A')}")
    print(f"   Address: {wallet_data.get('address', 'N/A')}")
    print(f"   Chain Type: {wallet_data.get('chain_type', 'N/A')}")

    # Test fetching the wallet
    if 'id' in wallet_data:
        wallet_id = wallet_data['id']
        print(f"\n2. Fetching wallet by ID: {wallet_id}...")

        get_url = f'https://api.privy.io/v1/wallets/{wallet_id}'
        get_headers = {
            'Authorization': f'Basic {credentials}',
            'privy-app-id': app_id
        }

        get_response = requests.get(get_url, headers=get_headers)
        print(f"Status Code: {get_response.status_code}")
        print(f"Response: {json.dumps(get_response.json(), indent=2)}")

        if get_response.status_code == 200:
            print("\n✅ Wallet fetched successfully!")
else:
    print("❌ Stellar wallet creation failed!")

File: test-privy-stellar.php
<?php
// Test Privy Stellar Wallet Creation
require_once __DIR__ . '/backend/config/privy.php';

echo "Testing Privy Stellar Wallet Creation...\n\n";

$privyAuth = new PrivyAuth();

// Test: Create a Stellar wallet
echo "1. Creating Stellar wallet via Privy...\n";
$testUserId = 'test_stellar_' . time();
$result = $privyAuth->createStellarWallet($testUserId);

if ($result && isset($result['address'])) {
    echo "✅ Stellar wallet created successfully!\n";
    echo "   Wallet ID: " . ($result['id'] ?? 'N/A') . "\n";
    echo "   Address: " . $result['address'] . "\n";
    echo "   Chain Type: " . ($result['chain_type'] ?? 'N/A') . "\n";
    echo "   Owner ID: " . ($result['owner_id'] ?? 'N/A') . "\n\n";

    // Test: Get the wallet by ID
    if (isset($result['id'])) {
        echo "2. Fetching wallet by ID...\n";
        $walletId = $result['id'];
        $fetchedWallet = $privyAuth->getStellarWallet($walletId);

        if ($fetchedWallet && isset($fetchedWallet['address'])) {
            echo "✅ Wallet fetched successfully!\n";
            echo "   Address matches: " . ($fetchedWallet['address'] === $result['address'] ? "Yes" : "No") . "\n";
        } else {
            echo "❌ Failed to fetch wallet\n";
        }
    }
} else {
    echo "❌ Stellar wallet creation failed!\n";
    echo "Response: " . json_encode($result, JSON_PRETTY_PRINT) . "\n";
}
